lumicks/pylake/correlated_stack.py

```diff
@@ -17,10 +21,59 @@ class TiffFrame:
     def __init__(self, page):
         self._src = page
 
+    def align(self):
+        """ reconstruct image using alignment matrices from Bluelake 
+            return aligned image as a NumPy array 
+        """
+
+        def correct_alignment_offset(alignment, x_offset, y_offset):
+            # translate the origin of the image so that it matches that of the original transform
+            transform1 = np.eye(3)
+            transform1[0,-1] = -x_offset
+            transform1[1,-1] = -y_offset
+            # apply the original transform to the translated image. 
+            # it only needs to be resized from a 2x3 to a 3x3 matrix
+            transform2 = np.vstack((alignment, [0,0,1]))
+            # translate the image back to the original origin. 
+            # takes into account both the offset and the scaling performed by the first step
+            transform3 = np.eye(3)
+            transform3[0,-1] = transform2[0,0]*x_offset
+            transform3[1,-1] = transform2[1,1]*y_offset
+            # concatenate the transforms by multiplying their matrices and ignore the unused 3rd row
+            return np.dot(transform3, np.dot(transform2, transform1))[:2,:]
+
+
+        description = json.loads(self._src.description)
```

Comment:
Perhaps just a warning that only the unaligned data is available? It might not be ideal for visualization, but as of right now there are no other methods I can see that depend on the presence of the aligned data.

Reply:
Yeah, considering that after discussing with Roeland we concluded that aligned should be the default, it should probably not warn when alignment is not possible, since you'd be getting that warning a lot on older datasets. This function should probably only be called if aligned data is available. Could even consider making this function private (at least initially), since if we do it lazily, we don't really need users to explicitly call it.